java-jsp/src/main/java/org/sonar/java/jsp/Jasper.java

```diff
@@ -33,38 +35,71 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
-import org.apache.jasper.JspC;
+import javax.servlet.jsp.JspFactory;
+import org.apache.jasper.JasperException;
+import org.apache.jasper.JspCompilationContext;
+import org.apache.jasper.compiler.Compiler;
+import org.apache.jasper.compiler.JspRuntimeContext;
+import org.apache.jasper.runtime.JspFactoryImpl;
+import org.apache.jasper.servlet.JspCServletContext;
 import org.sonar.api.batch.fs.FileSystem;
 import org.sonar.api.batch.fs.InputFile;
 import org.sonar.api.batch.sensor.SensorContext;
 import org.sonar.api.scanner.ScannerSide;
 import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
-import org.sonar.java.AnalysisException;
 import org.sonar.java.model.GeneratedFile;
 import org.sonar.java.model.SmapFile;
 
-import static java.util.Arrays.asList;
-
 @ScannerSide
 public class Jasper {
 
   private static final Logger LOG = Loggers.get(Jasper.class);
 
-  public Collection<GeneratedFile> generateFiles(SensorContext context, List<File> javaClasspath) {
-    List<Path> jspFiles = jspFiles(context.fileSystem());
+  public Collection<GeneratedFile> generateFiles(SensorContext sensorContext, List<File> javaClasspath) {
```

Comment:
cosmetic but I tend to think that this method could be made a bit more readable by extracting the core of the loop into a compileJspFile(... ) method